### PR TITLE
Fix retry mechanism for APNS

### PR DESCRIPTION
**Pull request by miguelreiswildlife**

### CODE REVIEW

File: extensions/apns_message_handler.go [hunk 1]
```diff
@@ -165,7 +173,7 @@ func (a *APNSMessageHandler) loadConfigurationDefaults() {
 // HandleResponses from apns.
 func (a *APNSMessageHandler) HandleResponses() {
 	for response := range a.PushQueue.ResponseChannel() {
-		a.handleAPNSResponse(response)
+		_ = a.handleAPNSResponse(response)
```
Comment (gmurayama) [inline extensions/apns_message_handler.go:176]: Is there a reason why we are ignoring the return of this function?

File: feedback/invalid_token.go [hunk 2]
```diff
@@ -145,30 +144,33 @@ func (i *InvalidTokenHandler) processMessages() {
 	flushTicker := time.NewTicker(i.flushTime)
 	defer flushTicker.Stop()
 
-	for i.run {
+	for {
 		select {
 		case tk, ok := <-i.InChan:
 			if ok {
+				i.BufferLock.Lock()
```
Comment (gmurayama) [inline feedback/invalid_token.go:151]: Nice catch!

File: pusher/pusher.go [hunk 3]
```diff
@@ -109,21 +122,17 @@ func (p *Pusher) Start(ctx context.Context) {
 		go v.CleanMetadataCache()
 	}
 	//nolint[:errcheck]
-	go p.Queue.ConsumeLoop()
-	go p.reportGoStats()
+	go p.Queue.ConsumeLoop(ctx)
+	//go p.reportGoStats()
```
Comment (gmurayama) [inline pusher/pusher.go:126]: Is this intentional?

**Review by gmurayama — DISMISSED** [review]
Overall LGTM

File: extensions/apns_message_handler.go (hunk 1 above)
Comment (miguelreiswildlife) [reply extensions/apns_message_handler.go:176]: We were already ignoring. I added a `_` just so Goland would stop giving warnings. I'll add a log as well, we don't really have anything to do if that fails.

**Review by gmurayama — APPROVED** [review]

**Review by gmurayama — APPROVED** [review]

**Pull request merged**